tools/istio-iptables/pkg/builder/iptables_builder_test.go

```diff
@@ -16,31 +16,70 @@ package builder
 
 import (
 	"reflect"
+	"strings"
 	"testing"
 
 	"istio.io/istio/tools/istio-iptables/pkg/config"
 	"istio.io/istio/tools/istio-iptables/pkg/constants"
 	iptableslog "istio.io/istio/tools/istio-iptables/pkg/log"
 )
 
-// TODO(abhide): Add more testcases once BuildV6Restore() are implemented
-func TestBuildV6Restore(t *testing.T) {
-	iptables := NewIptablesRuleBuilder(nil)
-	expected := ""
-	actual := iptables.BuildV6Restore()
-	if expected != actual {
-		t.Errorf("Output didn't match: Got: %s, Expected: %s", actual, expected)
-	}
-}
+// getMapFromRestore function takes a string in iptables-restore format and returns a map of the tables, chains, and rules.
```

Comment:
What's the benefit of doing this over the `goldens` approach we use in the other tests?

That is, just `Restore`-ing and comparing the string result to an expected result file?

Reply:
_editing my response since I misunderstood the question._

Technically there is no advantage over the goldens, I had the parser ready due to the other iptables commit.

I can update the tests to use the golden although I need to check first that the output of BuildV46Restore always returns the string the same way and it is not influenced by the internal tableRulesMap not being ordered.
Let me take a look

Reply:
I think the iteration of https://github.com/istio/istio/blob/master/tools/istio-iptables/pkg/builder/iptables_builder_impl.go#L183 doest not provide a gurantee on the order which could lead to different restore output.
I will add a sorting and update the PR to use goldens file for both old and new tests